src/action/updateRoleEnvironmentAccess.ts

```diff
@@ -8,9 +8,27 @@ export default async function ({ space, environment }: { space: Space; environme
   try {
     Logger.log(`Updating the Role ${roleId} to have access on Environment: ${environment.sys.id}.`);
     const role = await space.getRole(roleId);
-    role.policies.push(createAllowPolicyForEnvironment(environment.sys.id));
-    await role.update();
-    Logger.success(`Successfully updated the Role ${roleId} with Environment ${environment.sys.id}.`);
+
+    let isEnvironmentIdInRolePolicy = false;
```

Comment:
can we do something like this? 

```
const hasEnvironmentAccess = role.policies.some(policy =>
  policy.effect === 'allow' &&
  Array.isArray(policy.actions) &&
  policy.actions.includes('access') && // No need for @ts-expect-error if actions can include 'access'
  containsEnvironmentId(policy.constraint, environment.sys.id)
);

```

Reply:
can't we simplify this by just checking for `containsEnvironmentId`? At the end of the day, I don't think we need to check `policy.effect === 'allow' &&
  Array.isArray(policy.actions) &&
  policy.actions.includes('access')`.

Reply:
As a final suggestion, inverting the logic and returning first is a good practice, the final code should look something like this: 
`export default async function({ space, environment }: { space: Space; environment: Environment }) {
  const roleId: string = INPUT_UPDATE_ENVIRONMENT_ACCESS_ROLE_ID!;
  try {
    Logger.log(`Updating the Role ${roleId} to have access on Environment: ${environment.sys.id}.`);
    const role = await space.getRole(roleId);
    const isEnvironmentIdInRolePolicy = role.policies.some(policy => containsEnvironmentId(policy.constraint, environment.sys.id));
    if (isEnvironmentIdInRolePolicy) {
      Logger.log(`The Role ${roleId} has already been given access to Environment ${environment.sys.id}.`);
      return;
    }
    role.policies.push(createAllowPolicyForEnvironment(environment.sys.id));
    await role.update();
    Logger.success(`Successfully updated the Role ${roleId} with Environment ${environment.sys.id}.`);
  } catch (error) {
    Logger.error(`Failed to update the Role ${roleId} with Environment ${environment.sys.id}.`);
    Logger.verbose(error);
  }
}
`

Reply:
i included the checks for `policy.effect` and `policy.actions` in the off chance the EE was manually created in Contentful's UI with the same name as the feature branch, but given different `policy.effect` and `policy.actions` values. this scenario will probably never happen, so i would say it is safe to not include. figured to include it and have the reviewer determine if its necessary

Reply:
using a one-liner to iterate through the array is a lot cleaner, and i agree the if logic should not start with bang operator. will update to reflect those changes

Reply:
update: your suggestion had a `return` in the if statement, but lint rule asks for a padding-line-between-statements and prettier rule asks to delete the line. instead i wrapped everything after the if statement in an else statement.

Reply:
Why not just add that line that the linter complains about?
Prematurely returning is much cleaner then if/else

Reply:
Also https://www.samanthaming.com/tidbits/23-no-else-return/